recipe/build.sh

```diff
@@ -24,11 +24,38 @@ else
   export FORCE_CUDA=0
 fi
 
-# export USE_MKL_BLAS=1  # only used for >0.1.0
-export Torch_DIR=$(python -c 'import torch; print(torch.utils.cmake_prefix_path)')
+# Dynamic libraries need to be lazily loaded so that torch
+# can be imported on system without a GPU
+export LDFLAGS="${LDFLAGS//-Wl,-z,now/-Wl,-z,lazy}"
 
+# export USE_MKL_BLAS=1  # only used for >0.1.0
 export FORCE_NINJA=1
 export EXTERNAL_PHMAP_INCLUDE_DIR="${BUILD_PREFIX}/include/"
 export EXTERNAL_CUTLASS_INCLUDE_DIR="${BUILD_PREFIX}/include/"
 
-python -m pip install . -vvv
+export PYG_CMAKE_ARGS="${CMAKE_ARGS}"
+
+# get torch libraries for osx-arm64
+# from https://github.com/conda-forge/openmm-torch-feedstock/blob/f7b09cd93f69d7213acd88dca1b0b1770b0ac2bc/recipe/build.sh#L7
+LIBTORCH_DIR=${BUILD_PREFIX}
+if [[ "$OSTYPE" == "darwin"* && $OSX_ARCH == "arm64" ]]; then
+
+  LIBTORCH_DIR=${RECIPE_DIR}/libtorch
+  conda list -p ${BUILD_PREFIX} >packages.txt
+  cat packages.txt
+  PYTORCH_PACKAGE_VERSION=$(grep pytorch packages.txt | awk -F ' ' '{print $2}')
+  CONDA_SUBDIR=osx-arm64 conda create -y -p ${LIBTORCH_DIR} --no-deps pytorch=${PYTORCH_PACKAGE_VERSION} python=${PY_VER}
```

Comment:
This looks like quite the hack. I'm not aware of any other pytorch-dependent package to need to drop down to installing something in the build scripts, and I doubt that this feedstock is special enough to need it.

Reply:
I agree. It's coming from https://github.com/conda-forge/openmm-torch-feedstock/blob/f7b09cd93f69d7213acd88dca1b0b1770b0ac2bc/recipe/build.sh#L7.

I looked at many other repos but couldn't find similar enough ones to pyg-lib. Only this hack works.

Can you think of something else? I agree it looks overcomplicated for cross compiling a torch package.

Reply:
@conda-forge/openmm-torch @conda-forge/pytorch-cpu 
I think we should come up with a better way here (and generally) to use pytorch in cross-compilation, and help pytorch-dependent feedstocks to do so.

FWIW, the openmm-bits where this is taken from were added in https://github.com/conda-forge/openmm-torch-feedstock/pull/33 without much visible discussion, and I would have said the same thing there:

Calling conda in build scripts is something that should never[^1] be necessary.

[^1]: There's at most a handful legitimate case in all of conda-forge; those don't apply here.

Reply:
It looks to me as though this should be in the build section (it is!) and then we need to figure out where that is to set the correct `Torch_DIR`.  Though this may have been tried in one of the many labor intensive iterations to get here.  And if that doesn't work but this manual installation does, that is very much a mystery.

Reply:
I tried that already. If someone wants to give it a new shot, I am all good. I am a bit clueless about what to try next on my side here.

Reply:
If someone has an example of a package with the same constraints, I could take a look and try to adapt it to that feedstock.

Reply:
@jakirkham any idea on this?

To summarize: pyg needs torch from build but torch from env during linking and it's not clear what is the best way to achieve this without the hack above. The build is CMake based.

Reply:
@h-vetinari I think I got it to work. Can you review https://github.com/conda-forge/pyg-lib-feedstock/pull/23 ?